[Bug]: Bundled hooks missing in OpenClaw 2026.2.2-3

Bundled (built-in) hooks are missing after upgrading to OpenClaw 2026.2.2-3.
The same setup shows bundled hooks correctly in 2026.2.1, but openclaw hooks list returns no hooks in 2026.2.2-3.

⸻

Steps to reproduce
	1.	Install OpenClaw 2026.2.1

npm install -g openclaw@2026.2.1
openclaw hooks list


	2.	Upgrade to OpenClaw 2026.2.2-3

npm install -g openclaw@2026.2.2-3
openclaw hooks list



⸻

Expected behavior

Bundled hooks shipped with OpenClaw should be automatically discovered and listed.

⸻

Actual behavior

openclaw hooks list shows:

No hooks found.


⸻

Environment
	•	Clawdbot version: 2026.2.1 → 2026.2.2-3
	•	OS: macOS (Apple Silicon)
	•	Install method: npm install -g openclaw

The workaround doesn't work for npm installs, hooks are loaded but fail during firing. The npm has coded file names e.g. deps-XYZY.js not deps.js

```
2026-02-05T02:14:13.238Z Failed to load hook boot-md: Cannot find module '/opt/homebrew/lib/node_modules/openclaw/dist/cli/deps.js' imported from /Users/xx/.openclaw/hooks/boot-md/handler.ts
2026-02-05T02:14:13.246Z Failed to load hook session-memory: Cannot find module '/Users/xx/agents/agent-scope.js' imported from /Users/xx/.openclaw/hooks/session-memory/handler.ts
```

> The workaround doesn't work for npm installs, hooks are loaded but fail during firing. The npm has coded file names e.g. deps-XYZY.js not deps.js
> 
> ```
> 2026-02-05T02:14:13.238Z Failed to load hook boot-md: Cannot find module '/opt/homebrew/lib/node_modules/openclaw/dist/cli/deps.js' imported from /Users/xx/.openclaw/hooks/boot-md/handler.ts
> 2026-02-05T02:14:13.246Z Failed to load hook session-memory: Cannot find module '/Users/xx/agents/agent-scope.js' imported from /Users/xx/.openclaw/hooks/session-memory/handler.ts
> ```

I can confirm that. This workaround doesn't work with my bash script install (which was a npm install). I remove all the packages and reinstalled the OpenClaw 2026.2.1 (ed4529e)

> > The workaround doesn't work for npm installs, hooks are loaded but fail during firing. The npm has coded file names e.g. deps-XYZY.js not deps.js
> > ```
> > 2026-02-05T02:14:13.238Z Failed to load hook boot-md: Cannot find module '/opt/homebrew/lib/node_modules/openclaw/dist/cli/deps.js' imported from /Users/xx/.openclaw/hooks/boot-md/handler.ts
> > 2026-02-05T02:14:13.246Z Failed to load hook session-memory: Cannot find module '/Users/xx/agents/agent-scope.js' imported from /Users/xx/.openclaw/hooks/session-memory/handler.ts
> > ```
> 
> I can confirm that. This workaround doesn't work with my bash script install (which was a npm install). I remove all the packages and reinstalled the OpenClaw 2026.2.1 ([ed4529e](https://github.com/openclaw/openclaw/commit/ed4529e24673fb19ea506bb04b2c6d3deed6a451))

how to install 2026.2.1 for ’curl -fsSL https://openclaw.ai/install.sh | bash‘
thanks

> > > The workaround doesn't work for npm installs, hooks are loaded but fail during firing. The npm has coded file names e.g. deps-XYZY.js not deps.js
> > > ```
> > > 2026-02-05T02:14:13.238Z Failed to load hook boot-md: Cannot find module '/opt/homebrew/lib/node_modules/openclaw/dist/cli/deps.js' imported from /Users/xx/.openclaw/hooks/boot-md/handler.ts
> > > 2026-02-05T02:14:13.246Z Failed to load hook session-memory: Cannot find module '/Users/xx/agents/agent-scope.js' imported from /Users/xx/.openclaw/hooks/session-memory/handler.ts
> > > ```
> > 
> > 
> > I can confirm that. This workaround doesn't work with my bash script install (which was a npm install). I remove all the packages and reinstalled the OpenClaw 2026.2.1 ([ed4529e](https://github.com/openclaw/openclaw/commit/ed4529e24673fb19ea506bb04b2c6d3deed6a451))
> 
> how to install 2026.2.1 for ’curl -fsSL https://openclaw.ai/install.sh | bash‘ thanks

I didn't. i just "npm install -g openclaw@2026.2.1". And remember to backup your ~/.openclaw folder

--- a/scripts/copy-hook-metadata.ts
+++ b/scripts/copy-hook-metadata.ts
@@ -1,6 +1,6 @@
 #!/usr/bin/env tsx
 /**
- * Copy HOOK.md files from src/hooks/bundled to dist/hooks/bundled
+ * Copy HOOK.md files from src/hooks/bundled to dist/bundled
  */
 
 import fs from "node:fs";
@@ -11,7 +11,7 @@ const __dirname = path.dirname(fileURLToPath(import.meta.url));
 const projectRoot = path.resolve(__dirname, "..");
 
 const srcBundled = path.join(projectRoot, "src", "hooks", "bundled");
-const distBundled = path.join(projectRoot, "dist", "hooks", "bundled");
+const distBundled = path.join(projectRoot, "dist", "bundled");
 
 function copyHookMetadata() {
   if (!fs.existsSync(srcBundled)) {

--- a/tsdown.config.ts
+++ b/tsdown.config.ts
@@ -36,4 +36,18 @@ export default defineConfig([
     fixedExtension: false,
     platform: "node",
   },
+  // Bundled hook handlers – these were missed when migrating from tsc to tsdown
+  // in 2026.2.2, causing `openclaw hooks list` to show 0 hooks on npm installs.
+  {
+    entry: {
+      "hooks/bundled/session-memory/handler": "src/hooks/bundled/session-memory/handler.ts",
+      "hooks/bundled/command-logger/handler": "src/hooks/bundled/command-logger/handler.ts",
+      "hooks/bundled/boot-md/handler": "src/hooks/bundled/boot-md/handler.ts",
+      "hooks/bundled/soul-evil/handler": "src/hooks/bundled/soul-evil/handler.ts",
+      "hooks/llm-slug-generator": "src/hooks/llm-slug-generator.ts",
+    },
+    env,
+    fixedExtension: false,
+    platform: "node",
+  },
 ]);

--- a/tsdown.config.ts
+++ b/tsdown.config.ts
@@ -36,16 +36,8 @@ export default defineConfig([
     fixedExtension: false,
     platform: "node",
   },
-  // Bundled hook handlers – these were missed when migrating from tsc to tsdown
-  // in 2026.2.2, causing `openclaw hooks list` to show 0 hooks on npm installs.
   {
-    entry: {
-      "hooks/bundled/session-memory/handler": "src/hooks/bundled/session-memory/handler.ts",
-      "hooks/bundled/command-logger/handler": "src/hooks/bundled/command-logger/handler.ts",
-      "hooks/bundled/boot-md/handler": "src/hooks/bundled/boot-md/handler.ts",
-      "hooks/bundled/soul-evil/handler": "src/hooks/bundled/soul-evil/handler.ts",
-      "hooks/llm-slug-generator": "src/hooks/llm-slug-generator.ts",
-    },
+    entry: ["src/hooks/bundled/*/handler.ts", "src/hooks/llm-slug-generator.ts"],
     env,
     fixedExtension: false,
     platform: "node",

